p2p/discovery/ping.go

```diff
@@ -17,18 +18,19 @@ func (p *protocol) newPingRequestHandler() func(msg server.Message) []byte {
 	return func(msg server.Message) []byte {
 		plogger := p.logger.WithFields(log.String("type", "ping"), log.String("from", msg.Sender().String()))
 		plogger.Debug("handle request")
+		plogger.Warning("In ping request handler")
 		pinged := &node.Info{}
 		err := types.BytesToInterface(msg.Bytes(), pinged)
 		if err != nil {
 			plogger.Error("failed to deserialize ping message err=", err)
 			return nil
 		}
-
 		if err := p.verifyPinger(msg.Metadata().FromAddress, pinged); err != nil {
 			plogger.Error("msg contents were not valid err=", err)
 			return nil
 		}
-
+		//only update the last pinged time once we know everything is valid
+		p.lastDiscoveryPing = time.Now()
```

Comment:
is this used for anything other than tests?

Reply:
No, I was trying to figure out a better way to test if a peer has been pinged without adding this flag to the protocol that is only used for tests. Any thoughts?

Reply:
it could probably be done by mocking out some part of the function like `verifyPinger` and making it run specific test logic and mutate test state. there might be other parts that could be used to achieve the same thing.

Reply:
Pushed this logic down to the sim layer, when a message is being sent